### ISSUE TITLE
Channel <-> Event Binding

### ISSUE THREAD

**Issue by wayfarerx** [ESORaidplanner/ESORaidplanner-discord#23]
I would be cool if a GM could say:

`!bind <event-id>` in a channel

and make it so that all further Discord commands don't need an event ID, you could just:

`!signup h Templar` or `!signup "Heals-with-Song"`

and that would sign you up for the next instance of `<event-id>`.

**Comment by wayfarerx** [comment]
Some thoughts on this:

- This should actually be `!bind <recurring-event-id>`. This would mean that we should either add a Discord command that lists the recurring events & IDs (i.e. `!recurring`) or that recurring event IDs should be more prominent on esoraidplanner.com (or both TBH).

- We'd need an `!unbind` command as well.

- The bot would need to know what channels are bound to what recurring event ID. This should probably be optionally included in the response from `/api/discord/last-activity` though I have other ideas if that is too onerous.

- We'd need a way to map from `<recurring-event-id>` to `<event-id>`. We could either:
  - Have your service handle it, meaning endpoints that accept an `event_id` would need to also accept `recurring_event_id` in its place, mapping it to the next instance of the recurring event.
  - ...or provide an endpoint that takes a `recurring_event_id` and returns the `event_id` of the next instance of the recurring event.

**Comment by wayfarerx** [comment]
- Oh and the `!help` command output should take bindings into account:
  - On an unbound channel it would say `!signup [event-id] [class] [role]`
  - On a bound channel just `!signup [class] [role]`

That would hold for all commands with a `[event-id]` parameter.

**Comment by Woeler** [comment]
I am reworking the entire core of eso-raidplanner. I feel like it would make more sense to add that in there once it's done, as with the v1 core it would require me to make a mess.

**Comment by wayfarerx** [comment]
Yup absolutely no rush just war gaming this out in my head and wanted to put down my thoughts for posterity.

**Comment by wayfarerx** [comment]
Okay still not rushing you just using this as a notepad for ideas when they hit me.

@MissBizz runs a weekly event, but it alternates on a biweekly basis from one team to another. She's not excited to have two signup channels for the two alternating events. But wait, what if we could bind a Discord channel to MULTIPLE recurring events?!

Basically the same as above but `[event-id]` becomes optional if a channel has one or more event bindings. `!bind` would accept zero or more recurring event IDs, as would `!unbind` (we can debate the behavior of `!bind` and `!unbind` when they are called with no parameters).

Other commands that can optionally take an `[event-id]` will, when an `[event-id]` is not provided, use the event ID of the next upcoming event instance selected from all the bound recurring events (or get angry if the channel is not bound).

tl;dr you have one Discord channel bound to multiple recurring events. If people `!signup` or `!signoff` or whatever it applies it to the next upcoming event selected from all the bound recurring events. So if `Trials (backup)` is this coming Saturday and `Trials (main)` is the following Saturday when Wayfarer signs up with `!signup "Heals-with-Song"` I've actually signed up for this coming Saturday's `Trials (backup)` team because it is happening sooner than `Trials (main)`. It's up to @MissBizz to look and say "Oh Wayfarer is on the backup team so he gets a priority position in the backup team," or vice-versa.
